### PR TITLE
Cb/struct update parsing

### PR DESCRIPTION
**Pull request by claytonpbarrows**
this PR resolves issues with parsing data into updated struct definitions

### CODE REVIEW

File: src/parsers/common.jl [hunk 1]
```diff
@@ -80,3 +80,43 @@ function convert_units!(value::Float64,
     end
     return value
 end
+
+function Base.convert(::Type{ThermalFuels}, fuel::AbstractString)
+    map = ((string(e),e) for e in instances(ThermalFuels)) |> Dict
```
Comment (daniel-thom) [inline src/parsers/common.jl]:
```suggestion
    map = Dict(string(e) => e) for e in instances(ThermalFuels))
```

File: src/parsers/common.jl [hunk 2]
```diff
@@ -80,3 +80,43 @@ function convert_units!(value::Float64,
...
+    extras = Dict("NG" => NATURAL_GAS::ThermalFuels,
+                  "NUC" => NUCLEAR::ThermalFuels,
+                  "GAS" => NATURAL_GAS::ThermalFuels,
+                  "OIL" => DISTILLATE_FUEL_OIL::ThermalFuels,
+                  "SYNC_COND" => OTHER::ThermalFuels)
+
+    [push!(map, e) for e in extras]
```
Comment (daniel-thom) [inline src/parsers/common.jl]: Can we build this dict one time instead of every time?

File: src/validation/validation_functions.jl [hunk 3]
```diff
@@ -23,7 +23,8 @@ function get_field_descriptor(struct_descriptor::Dict, fieldname::AbstractString
             return field
         end
     end
-    error("field $fieldname does not exist in $(struct_descriptor["name"])")
+
+    error("field $fieldname does not exist in $(struct_descriptor["struct_name"]) validation config")
```
Comment (jd-lara) [inline src/validation/validation_functions.jl]: Throw DataError?

File: src/validation/validation_functions.jl [hunk 4]
```diff
@@ -102,14 +123,16 @@ function check_limits(::Type{T}, valid_info::ValidationInfo, field_value) where
     @assert length(field_value) == 2
     result1 = check_limits_impl(valid_info, field_value[1])
     result2 = check_limits_impl(valid_info, field_value[2])
-    return result1 && result2
+    return  result1 && result2
```
Comment (jd-lara) [inline src/validation/validation_functions.jl]: Extra space

**Pull request merged**